### PR TITLE
Revamping the security structure and CMEK layout to accomodate for additions in future

### PR DESCRIPTION
**Pull request by fiquick**

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption at rest](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Secure connections](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/secure-connections)
- [Single sign-on](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption at rest](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Single sign-on](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption at rest](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Single sign-on](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption at rest](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Single sign-on](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption at rest](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Single sign-on](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption at rest](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Single sign-on](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption at rest](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Single sign-on](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

File: modules/ROOT/content-nav.adoc [hunk 1]
```diff
@@ -7,12 +7,16 @@ Generic Start
 
 * xref:platform/create-account.adoc[]
 * xref:platform/cloud-providers.adoc[Cloud provider marketplaces]
-* xref:platform/security.adoc[]
 * xref:platform/user-management.adoc[]
 * xref:platform/apoc.adoc[]
 * xref:platform/logging.adoc[Logging]
 
-* Neo4j Connectors
+* Neo4j security
```
Comment (AlexicaWright) [inline modules/ROOT/content-nav.adoc:14]:
```suggestion
* Security
```

File: modules/ROOT/pages/platform/security/encryption-at-rest.adoc [hunk 2]
```diff
@@ -0,0 +1,99 @@
+[[aura-reference-security]]
+= Encryption at rest
```
Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security/encryption-at-rest.adoc:2]:
```suggestion
= Encryption
```
Unless we also have sections on _Encryption in transit_ and _End-to-end encryption_, what is the need to specify?

File: modules/ROOT/pages/platform/security/encryption-at-rest.adoc [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+:description: Aura is encrypted using intra-cluster encryption, and is CMK compatible.
+
+All data stored in Neo4j Aura is encrypted using intra-cluster encryption between the various nodes comprising your instance and encrypted at rest using the underlying cloud provider's encryption mechanism.
+
+By default, each cloud provider encrypts all backup buckets (including the objects stored inside) using either https://cloud.google.com/storage/docs/encryption/default-keys[Google-managed encryption], https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingServerSideEncryption.html[AWS SSE-S3 encryption], or https://learn.microsoft.com/en-us/azure/storage/common/storage-service-encryption[Azure Storage encryption].
```
Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security/encryption-at-rest.adoc:7]:
```suggestion
By default, each cloud provider encrypts all backup buckets (including the objects stored inside) using either link:https://cloud.google.com/storage/docs/encryption/default-keys[Google-managed encryption], link:https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingServerSideEncryption.html[AWS SSE-S3 encryption], or link:https://learn.microsoft.com/en-us/azure/storage/common/storage-service-encryption[Azure Storage encryption].
```

File: modules/ROOT/pages/platform/security/encryption-at-rest.adoc [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+== Customer Managed Keys
+
+label:AuraDB-Enterprise[]
+label:AuraDS-Enterprise[]
+
+[CAUTION]
```
Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security/encryption-at-rest.adoc:14]:
```suggestion
[NOTE]
```

File: modules/ROOT/pages/platform/security/single-sign-on.adoc [hunk 5]
```diff
@@ -0,0 +1,53 @@
+[[aura-reference-security]]
+= Single sign-on
```
Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security/single-sign-on.adoc:2]: This section needs to be removed from the secure-connection.adoc file

Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security/single-sign-on.adoc:2]:
```suggestion
= Single Sign-On (SSO)
```

**Review by AlexicaWright — COMMENTED** [review]
Great work! Just a few thoughts and suggestions.

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption at rest](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Single sign-on](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption at rest](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Single sign-on](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption at rest](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Single sign-on](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Single Sign-On (SSO)](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Single Sign-On (SSO)](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

File: modules/ROOT/partials/import-database.adoc [hunk 6]
```diff
@@ -39,7 +39,7 @@ If the database you want to upload is running an earlier version of Neo4j, pleas
 
 [CAUTION]
 ====
-The `neo4j-admin push-to-cloud` command in Neo4j 4.4 and earlier is not compatible with instances encrypted with xref:platform/security.adoc#_customer_managed_keys[Customer Managed Keys].
+The `neo4j-admin push-to-cloud` command in Neo4j 4.4 and earlier is not compatible with instances encrypted with xref:platform/security/encryption-at-rest.adoc#_customer_managed_keys[Customer Managed Keys].
```
Comment (AlexicaWright) [inline modules/ROOT/partials/import-database.adoc:42]: You may want to consider renaming this file to just _encryption.adoc_.

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Single Sign-On (SSO)](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption)
- [Single Sign-On (SSO)](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

File: modules/ROOT/partials/import-database.adoc (hunk 6 above)
Comment (fiquick) [reply modules/ROOT/partials/import-database.adoc:42]: I've renamed it!

File: modules/ROOT/pages/platform/security/encryption-at-rest.adoc [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+== AWS keys
+
+=== Creating an AWS key
+
+* Create a key in the AWS KMS ensuring the region matches your Aura database instance.
+Copy the generated ARN.
+You need it in the next step.
+* Go to *security settings* in the Aura Console, create a *Customer Managed Key* and copy the JSON code that is generated in the Aura Console when you add a key.
+* In the AWS KMS, edit the key policy to include the JSON code.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security/encryption-at-rest.adoc:75]:
```suggestion
. Create a key in the AWS KMS ensuring the region matches your Aura database instance.
Copy the generated ARN.
You need it in the next step.
. Go to *security settings* in the Aura Console, create a *Customer Managed Key* and copy the JSON code that is generated in the Aura Console when you add a key.
. In the AWS KMS, edit the key policy to include the JSON code.
```

File: modules/ROOT/pages/platform/security/encryption.adoc [hunk 8]
```diff
@@ -0,0 +1,99 @@
+[[aura-reference-security]]
+= Encryption
```
Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security/encryption.adoc:2]: What is the motivation behind two identical files with different names? The comments above apply in this file as well, whichever one you decide to keep.

File: modules/ROOT/pages/platform/security/single-sign-on.adoc [hunk 9]
```diff
@@ -0,0 +1,53 @@
+[[aura-reference-security]]
+= Single Sign-On (SSO)
+:description: SSO allows users to log in to the Aura Console using their company IdP credentials.
+
+label:AuraDB-Enterprise[]
+label:AuraDS-Enterprise[]
+
+Aura Enterprise supports Single Sign-On (SSO) at both the Console level and for accessing Workspace, Bloom and Browser clients directly at the instance level.
+
+[NOTE]
```
Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security/single-sign-on.adoc:10]: Why is this a note?

File: modules/ROOT/pages/platform/security/single-sign-on.adoc [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+====
+Accessing Aura with SSO requires:
+ 
+* Authorization Code Flow with PKCE.
+* A publicly accessible Identity Provider (IdP) server.
+====
+
+== Console SSO
+
+Console SSO allows users to log in to the Aura Console using their company IdP credentials and grants link:{neo4j-docs-base-uri}/cypher-manual/current/administration/access-control/built-in-roles#access-control-built-in-roles-public[Public Access privileges] to all instances in the tenant.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security/single-sign-on.adoc:20]: Third person plural here and then second person below and again in _Instance SSO_. Neither is wrong, but it should be consistent. 
_Console SSO allows users to..._ and below:
_Instance SSO allows you to..._ , is it possible to rewrite?

**Review by AlexicaWright — COMMENTED** [review]
Some comments :)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption)
- [Single Sign-On (SSO)](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption)
- [Single Sign-On (SSO)](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption)
- [Single Sign-On (SSO)](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption)
- [Single Sign-On (SSO)](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption)
- [Single Sign-On (SSO)](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption-at-rest)
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption)
- [Single Sign-On (SSO)](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption)
- [Single Sign-On (SSO)](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

File: modules/ROOT/pages/platform/security/encryption-at-rest.adoc [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+Manual key rotation is **not** recommended.
+
+=== Regionality
+
+There is a limit of one key for AuraDB and one key for AuraDS per region.
+Depending on the KMS, there may be a delay between disabling a key, and when it can no longer be used to encrypt and decrypt data.
+
+=== Importing an existing database
+
+To upload a database greater than 4GB to instances encrypted with Customer Managed Keys in Neo4j 5, use `neo4j-admin database upload`.
```
[the remark this answers is not in the document]
Comment (fiquick) [reply modules/ROOT/pages/platform/security/encryption-at-rest.adoc]: That works as well with 'neo4j-admin database upload'. For files smaller than 4GB they can also use the Console web UI. The issue the docs address is that the web ui has the 4GB limit. It's not unique to CMK. This explains how to To import a .dump file under 4GB: https://neo4j.com/docs/aura/auradb/importing/import-database/#_import_database

Which information do you suggest I should highlight in this section? @AlexicaWright

File: modules/ROOT/pages/platform/security/encryption.adoc [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+Manual key rotation is **not** recommended.
+
+=== Regionality
+
+There is a limit of one key for AuraDB and one key for AuraDS per region.
+Depending on the KMS, there may be a delay between disabling a key, and when it can no longer be used to encrypt and decrypt data.
+
+=== Importing an existing database
+
+To upload a database greater than 4GB to instances encrypted with Customer Managed Keys in Neo4j 5, use `neo4j-admin database upload`.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security/encryption.adoc:57]:
```suggestion
You can upload a database to instances encrypted with Customer Managed Keys in Neo4j 5 directly from the console or by using `neo4j-admin database upload`.
If the database is larger than 4 GB, you have to use `neo4j-admin database upload`. 
```

**Review by AlexicaWright — COMMENTED** [review]
Some more suggestions.

File: modules/ROOT/pages/platform/security/encryption.adoc (hunk 12 above)
Comment (fiquick) [reply modules/ROOT/pages/platform/security/encryption.adoc:57]: that's great! thank you

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption)
- [Single Sign-On (SSO)](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

File: modules/ROOT/pages/platform/security/encryption-at-rest.adoc (hunk 11 above)
[the remark this answers is not in the document]
Comment (fiquick) [reply modules/ROOT/pages/platform/security/encryption-at-rest.adoc]: Thanks!

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption)
- [Single Sign-On (SSO)](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption)
- [Single Sign-On (SSO)](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

File: modules/ROOT/pages/platform/security/secure-connections.adoc [hunk 13]
```diff
@@ -196,171 +197,4 @@ To enable private endpoints using Azure Private Link:
...
-
-You can override this behavior by selecting the Neo4j Managed Key when cloning the database.
-If there is no valid CMK for the destination region and product, the Neo4j Managed Key is used to encrypt the cloned instance.
-
-=== Importing an existing database
-
-To upload a database greater than 4GB to instances encrypted with Customer Managed Keys in Neo4j 5, use `neo4j-admin database upload`.
-The `neo4j-admin push-to-cloud` command in Neo4j v4.4 and earlier is not supported for instances encrypted with Customer Managed Keys.
-For more information see the xref:auradb/importing/import-database.adoc#_neo4j_admin_database_upload[Neo4j Admin `database upload`] documentation.
+Please see the https://learn.microsoft.com/en-us/azure/private-link/rbac-permissions#private-endpoint[Azure Documentation] for required roles and permissions.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security/secure-connections.adoc:200]:
```suggestion
Please see the link:https://learn.microsoft.com/en-us/azure/private-link/rbac-permissions#private-endpoint[Azure Documentation] for required roles and permissions.
```

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption)
- [Single Sign-On (SSO)](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-351.surge.sh

New pages:
- [Encryption](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/encryption)
- [Single Sign-On (SSO)](https://neo4j-docs-aura-351.surge.sh/aura/platform/security/single-sign-on)

**Review by AlexicaWright — APPROVED** [review]

**Pull request merged**